assets/javascripts/discourse/initializers/discourse-automation.js

```diff
@@ -29,6 +34,136 @@ function _initializeDiscourseAutomation(api) {
+
+      if (!isEmpty(destinationUrl)) {
+        cookie("destination_url", destinationUrl, { path: "/" });
+      }
+
+      // Add the userfields to the data
+      if (!isEmpty(userFields)) {
+        attrs.userFields = {};
+        userFields.forEach(
+          (f) => (attrs.userFields[f.get("field.id")] = f.get("value"))
+        );
+      }
+
+      attrs["customFields"] = this.customFields;
+
+      this.set("formSubmitted", true);
+      return User.createAccount(attrs).then(
+        (result) => {
+          if (this.isDestroying || this.isDestroyed) {
+            return;
+          }
+
+          this.set("isDeveloper", false);
+          if (result.success) {
+            // invalidate honeypot
+            this._challengeExpiry = 1;
+
+            // Trigger the browser's password manager using the hidden static login form:
+            const $hidden_login_form = $("#hidden-login-form");
+            $hidden_login_form
+              .find("input[name=username]")
+              .val(attrs.accountUsername);
+            $hidden_login_form
+              .find("input[name=password]")
+              .val(attrs.accountPassword);
+            $hidden_login_form
+              .find("input[name=redirect]")
+              .val(userPath("account-created"));
+            $hidden_login_form.submit();
+            return new Promise(() => {}); // This will never resolve, the page will reload instead
+          } else {
+            this.flash(
+              result.message || I18n.t("create_account.failed"),
+              "error"
+            );
+            if (result.is_developer) {
+              this.set("isDeveloper", true);
+            }
+            if (
+              result.errors &&
+              result.errors.email &&
+              result.errors.email.length > 0 &&
+              result.values
+            ) {
+              this.rejectedEmails.pushObject(result.values.email);
+            }
+            if (
+              result.errors &&
+              result.errors.password &&
+              result.errors.password.length > 0
+            ) {
+              this.rejectedPasswords.pushObject(attrs.accountPassword);
+            }
+            this.set("formSubmitted", false);
+            removeCookie("destination_url");
+          }
+        },
+        () => {
+          this.set("formSubmitted", false);
+          removeCookie("destination_url");
+          return this.flash(I18n.t("create_account.failed"), "error");
+        }
+      );
+    },
+  });
+
+  User.reopenClass({
+    createAccount(attrs) {
+      let data = {
+        name: attrs.accountName,
+        email: attrs.accountEmail,
+        password: attrs.accountPassword,
+        username: attrs.accountUsername,
+        password_confirmation: attrs.accountPasswordConfirm,
+        challenge: attrs.accountChallenge,
+        user_fields: attrs.userFields,
+        timezone: moment.tz.guess(),
+      };
+
+      if (attrs.customFields) {
```

Comment:
Is the change here to add `custom_fields` to the create account flow?

Reply:
Yes, that's correct. We need to provide `custom_fields` to the create account API to include custom fields in account creation.